components/customNodes.tsx

```diff
@@ -59,22 +61,28 @@ export const Producer = ({ data }: any) => {
         data-popover-trigger="hover"
         type="button"
         onClick={handleModalOpen}
-        className="w-[205px] h-[64px] bg-white rounded-md shadow-lg border-1 border-purple-200 flex items-center justify-between px-1"
+        className="w-[250px] h-[64px] bg-white rounded-lg shadow-lg border-1 border-purple-200 flex items-center justify-between px-1"
       >
         <IconGripVertical
           class="w-6 h-6 text-purple-100 cursor-grab"
           id="dragHandle"
         />
-        <img src={"/images/placeholder-icon.png"} className="w-[30px]" />
-        <a href={`/flow/${data.label.toLowerCase()}`}>
-          <div className={"flex flex-col"}>
+        <ProducerIcon />
+        <a
+          href={`/${encodeURIComponent(data.pipeline.audience.serviceName)}/${
+            encodeURIComponent(data.pipeline.audience.componentName)
+          }/${
+            data.pipeline.audience.operationType === 1 ? "consumer" : "producer"
```

Comment:
You can do this a little more neatly with the typescript enum like `OperationType[1].toLowerCase()`.